proto/v2/skaffold.proto

```diff
@@ -0,0 +1,816 @@
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+message StateResponse {
+    State state = 1;
+}
+
+message Response {
+    string msg = 1;
+}
+
+message Request {
+    string name = 1;
+}
+
+// `State` represents the current state of the Skaffold components
+message State {
+    BuildState buildState = 1;
+    DeployState deployState = 2;
+    reserved 3; // field 3 is obsolete
+    map<int32, PortEvent> forwardedPorts = 4;
+    StatusCheckState statusCheckState = 5;
+    FileSyncState fileSyncState = 6;
+    repeated DebuggingContainerEvent debuggingContainers = 7;
+    Metadata metadata = 8;
+    TestState testState = 9;
+}
+
+message Metadata {
+    BuildMetadata build = 1;
+    DeployMetadata deploy = 2;
+    TestMetadata test = 3;
+    // Additional key value pairs to describe the build pipeline
+    map<string, string> additional = 99;
+}
+
+message BuildMetadata {
+  message ImageBuilder {
+      BuilderType type = 1;
+      int32 count = 2;
+  }
+  int32 numberOfArtifacts = 1;
+  repeated ImageBuilder builders= 2;
+  BuildType type = 3;
+  // Additional key value pairs to describe the deploy pipeline
+  map<string, string> additional = 99;
+}
+
+// TestMetadata describes the test pipeline
+message TestMetadata {
+    message Tester {
+        TesterType type = 1;
+        int32 count = 2;
+    }
+    repeated Tester Testers = 1;
+}
+
+message DeployMetadata {
+    message Deployer {
+        DeployerType type = 1;
+        int32 count = 2;
+    }
+    repeated Deployer deployers = 1;
+    ClusterType cluster = 2;
+}
+
+// `BuildState` maps Skaffold artifacts to their current build states
+message BuildState {
+    // A map of `artifact name -> build-state`.
+    // Artifact name is defined in the `skaffold.yaml`.
+    // The `build-state` can be: <br>
+    // - `"Not started"`: not yet started <br>
+    // - `"In progress"`: build started <br>
+    // - `"Complete"`: build succeeded <br>
+    // - `"Failed"`: build failed
+    map<string, string> artifacts = 1;
+    bool autoTrigger = 2;
+    StatusCode statusCode = 3;
+}
+
+// `TestState` describes the current state of the test
+message TestState {
+    // Status of the current test
+    string status = 1;
+    // Teststate status code
+    StatusCode statusCode = 2;
+}
+
+// `DeployState` describes the status of the current deploy
+message DeployState {
+    string status = 1;
+    bool autoTrigger = 2;
+    StatusCode statusCode = 3;
+}
+
+// `StatusCheckState` describes the state of status check of current deployed resources.
+message StatusCheckState {
+    string status = 1;
+    // A map of `resource name -> status-check-state`. Where `resource-name` is the kubernetes resource name.
+    // The `status-check-state` can be <br>
+    // - `"Not started"`: indicates that `status-check` has just started. <br>
+    // - `"In progress"`: InProgress is sent after every resource check is complete. <br>
+    // - `"Succeeded"`:
+    // - `"Failed"`:
```

Comment:
Since these are codes, my 2¢ is to remove the spaces and camel case them so that they do not look like English.  Especially given the comment uses `InProgress`.

Reply:
Good catch, this is actually in inconsistency that exists in v1, as the state API uses "In Progress" as a status while the event API uses "InProgress" as a status